be/src/vec/exec/join/process_hash_table_probe_impl.h

```diff
@@ -207,883 +208,601 @@ void ProcessHashTableProbe<JoinOpType>::_pre_serialize_key(
                 mcol, *_join_context->_left_output_slot_flags, current_offset, last_probe_index,
-                probe_size, all_match_one, false));
+                probe_size, all_match_one, with_other_conjuncts));
     }
 
     output_block->swap(mutable_block.to_block());
 
+    if constexpr (with_other_conjuncts) {
+        return do_other_join_conjuncts(output_block, is_mark_join, multi_matched_output_row_count,
+                                       is_the_last_sub_block);
+    }
+
     return Status::OK();
 }
 
 template <int JoinOpType>
-template <bool need_null_map_for_probe, bool ignore_null, typename HashTableType>
-Status ProcessHashTableProbe<JoinOpType>::do_process_with_other_join_conjuncts(
-        HashTableType& hash_table_ctx, ConstNullMapPtr null_map, MutableBlock& mutable_block,
-        Block* output_block, size_t probe_rows, bool is_mark_join) {
-    auto& probe_index = *_join_context->_probe_index;
-    auto& probe_raw_ptrs = *_join_context->_probe_columns;
-    if (_build_block_rows.size() < probe_rows * PROBE_SIDE_EXPLODE_RATE) {
-        _probe_indexs.resize(probe_rows * PROBE_SIDE_EXPLODE_RATE);
-        _build_block_rows.resize(probe_rows * PROBE_SIDE_EXPLODE_RATE);
-        _build_block_offsets.resize(probe_rows * PROBE_SIDE_EXPLODE_RATE);
+Status ProcessHashTableProbe<JoinOpType>::do_other_join_conjuncts(
+        Block* output_block, bool is_mark_join, int multi_matched_output_row_count,
+        bool is_the_last_sub_block) {
+    // dispose the other join conjunct exec
+    auto row_count = output_block->rows();
+    if (!row_count) {
+        return Status::OK();
     }
 
-    using KeyGetter = typename HashTableType::State;
-    using Mapped = typename HashTableType::Mapped;
-    if constexpr (std::is_same_v<Mapped, RowRefListWithFlags>) {
-        constexpr auto probe_all =
-                JoinOpType == TJoinOp::LEFT_OUTER_JOIN || JoinOpType == TJoinOp::FULL_OUTER_JOIN;
-        KeyGetter key_getter(probe_raw_ptrs, _join_context->_probe_key_sz, nullptr);
-
-        if constexpr (ColumnsHashing::IsPreSerializedKeysHashMethodTraits<KeyGetter>::value) {
-            if (probe_index == 0) {
-                _pre_serialize_key(probe_raw_ptrs, probe_rows, _probe_keys);
+    SCOPED_TIMER(_join_context->_process_other_join_conjunct_timer);
+    int orig_columns = output_block->columns();
+    IColumn::Filter other_conjunct_filter(row_count, 1);
+    bool can_be_filter_all = false;
+    RETURN_IF_ERROR(VExprContext::execute_conjuncts(*_join_context->_other_join_conjuncts, nullptr,
+                                                    output_block, &other_conjunct_filter,
+                                                    &can_be_filter_all));
+
+    auto result_column_id = output_block->columns();
+    auto filter_column = ColumnVector<UInt8>::create();
+    if (can_be_filter_all) {
+        memset(other_conjunct_filter.data(), 0, row_count);
+    }
+    filter_column->get_data() = std::move(other_conjunct_filter);
```

Comment:
here do move, the can_be_filter_all is useless

Reply:
removed